pkg/validation/limits.go

```diff
@@ -959,6 +970,27 @@ func (o *Overrides) StreamRetention(userID string) []StreamRetention {
 	return o.getOverridesForUser(userID).StreamRetention
 }
 
+// RetentionHours returns the retention period for a given user.
+func (o *Overrides) RetentionHours(userID string, ls labels.Labels) string {
+	streamRetentions := o.getOverridesForUser(userID).StreamRetention
+
+	selectedRetention := o.getOverridesForUser(userID).RetentionPeriod
+	highestPriority := -1
+
+	if len(ls) > 0 {
```

Comment:
this check is to help using this same function for when you want to automatically gets the Tenant retention but have no labels available

Reply:
I think it'd be more idiomatic to return early if ls is empty and keep this loop in the top level.